src/ssh.ts

```diff
@@ -1,5 +1,41 @@
 const knownHostRegex = /(?:[a-zA-Z0-9-]+\.)+[a-zA-Z0-9]+/; // will match numerical IPs as well as domains/subdomains
 
+type ExecuteShellCommandFunction = (
+  commandToExecute: string
+) => Promise<string>;
+
+const resolveAbsolutePath = (path: string, basePath: string): string => {
+  if (path.startsWith("/") || path.startsWith("~/") || path === "~") {
+    return path;
+  }
+
+  return basePath.replace(/\/$/, "") + "/" + path;
```

Comment:
Is this replacing the last slash and then adding it again?

Reply:
@fedeci Yes, I do this to support both `dir/` and `dir` as parameters for the function

Reply:
nitpicky but I prefer this syntax for future maintainers
```suggestion
  return basePath + (basePath.endsWith("/") ? "" : "/") + path;
```